src/main/java/org/reactome/addlinks/referencecreators/KEGGReferenceCreatorHelper.java

```diff
@@ -57,22 +62,23 @@ private synchronized String createNewKEGGReferenceDatabase(String targetIdentifi
 	 * it will *create* a new ReferenceDatabase object.
 	 * @param keggIdentifier - The KEGG Identifier
 	 * @param keggPrefix - The prefix for the KEGG Identifier.
-	 * @return An array with two elements: the first element is the ReferenceDatabase (a DB_ID, though in some cases, it could be the name of the ReferenceDatabase); the second element is the Kegg identifier, which might be 
+	 * @return An array with two elements: the first element is the ReferenceDatabase (a DB_ID, though in some cases, it could be the name of the ReferenceDatabase); the second element is the Kegg identifier, which might be
 	 */
 	public synchronized String[] determineKeggReferenceDatabase(String keggIdentifier, String keggPrefix)
 	{
+		String identifier = keggIdentifier;
 		String targetDB = null;
 		// "vg:" and "ad:" aren't in the species list because they are not actually species. So that's why it's OK to check for them here, after
 		// the identifier has already been pruned.
-		if (keggIdentifier.startsWith("vg:"))
+		if (identifier.startsWith("vg:"))
 		{
 			targetDB = "KEGG Gene (Viruses)";
-			keggIdentifier = keggIdentifier.replaceFirst("vg:", "");
+			identifier = identifier.replaceFirst("vg:", "");
 		}
-		else if (keggIdentifier.startsWith("ag:"))
+		else if (identifier.startsWith("ag:"))
 		{
 			targetDB = "KEGG Gene (Addendum)";
```

Comment:
I know this isn't a new thing, but I'm curious about where this DB comes from. Does Addendum imply it is like a miscellaneous Kegg database? I tried searching for it on KEGG but nothing came up

Reply:
On this page: https://www.genome.jp/kegg/genes.html , the first table contains "

"Addendum    ag    PubMed-based collection of functionally characterized proteins"

You can search this specific database here: https://www.genome.jp/dbget-bin/www_bfind?ag